### DISCUSSION TITLE
The Fungus Among Us: Mycus management strategies.

### DISCUSSION THREAD

**Discussion by I-am-Erk** [CleverRaven/Cataclysm-DDA#49232] [Ideas]
This is a discussion and compilation thread for ideas for how to manage fungus more appropriately.

## Design goal
The mycus and fungal enemies should generally fill the "plague/stealth" niche. They should not be individually threatening or harmful in combat, and for the most part shouldn't even engage. Instead their strategy should be slow takeover and subterfuge.

### The problem
Fungal expansion is currently utterly unsustainable even after a few nerfs. Very little actively controls fungal spread.

### Recommended solution
Fungal expansion should be a prototype for faction influence: faction influence is a way of determining borders, patrols, and combat between factions. We don't need all that at the moment, but what we do need is the ability to limit fungal spread based on proximity to a fungal control center (tower or fungal bloom, eg).

I suggest this basically be done by drawing a heatmap at the overmap level around a fungal tower. This would set a meta value for each OMT inside the heatmap, `fungal_influence`. The influence should then determine how many fungaloids, spore clouds, and the like are allowed to be in the OMT, how quickly the fungalize function is allowed to operate in the tile, how many tiles are allowed to be fungalized and maybe other things (eg mycus monsters could have more hp and regenerate faster in high influence OMTs). Fungal mobs can still move freely into new OMTs regardless of influence, spore clouds can still fungalize individual tiles and things, but no new fungal mobs will form if the population cap is reached, and if influence is 0 the cap on number of fungalized tiles in an OMT should be very low, like 10 tiles or so.

When a fungal tower or other high level fungal node is destroyed, its death should cause the heatmap to be redrawn in the overmap: a tower special should only contribute to the heatmap in this context if it has a surviving fungal tower in it, so the death of the tower will cause fungus in the area to stop growing and make it possible to be destroyed.

Later on in development, we can look into making it possible for fungal agents to create new blooms in areas over a certain amount of influence, which then cause the heat map to redraw. Maybe it should require some influence but it has to be low? So you see blooms crop up on the edges of tower zones, and maybe it requires a certain number of fungalized tiles and monsters in the OMT first. This would mean we'd see fungaloids turning themselves into blooms out on the outskirts of an existing tower after it has been established a while. Then, blooms should be able to upgrade to towers over time and expand their influence again.
 - In this context, when we destroy a tower we should leave a single area of low influence in the site the tower used to be, so that fungaloids can potentially come in and recreate a bloom there if left unattended. However when you destroy a bloom, it should leave no influence, allowing you to eventually clear the infection completely.

There is more to discuss here but I'm dropping this out for consideration as I have to get back to work.

**Comment by KorGgenT** [comment]
It would also be cool to add a static amount of "area" that the fungus could expand to with additional "control towers" or whatever. This would definitely leverage the negative exponential nature of that kind of thing, and make it much more obvious and reasonable for a player to remove fungus from an area, while also giving a chance to have different fungus in a higher populated/denser faction area.

**Reply by I-am-Erk** [reply]
yeah that's kinda what I am getting at with
> Later on in development, we can look into making it possible for higher level fungal agents to create new blooms in areas over a certain amount of influence, which then cause the heat map to redraw.

The bit I am not sure on is how to limit this. Maybe it should require *some* influence but it has to be low? So you see blooms crop up on the edges of tower zones, and maybe it requires a certain number of fungalized tiles and monsters in the OMT first. So you see fungaloids turning themselves into blooms out on the outskirts of an existing tower after it has been established a while. Then, blooms should be able to upgrade to towers over time and expand their influence again.

**Reply by KorGgenT** [reply]
Right, fungus will still expand, it will just be much much slower, and actually possible to stop the spread.

**Reply by I-am-Erk** [reply]
Exactly, and if you see a bloom crop up and go torch it then you stop further spread. It creates an incentive to go after the towers, and once the tower is destroyed you're safe for a while.

We might want to consider leaving a little influence behind afterwards, so that a fungaloid could come along and create a bloom in the area if you leave it unattended.

**Reply by Malorn-Deslor** [reply]
Sounds like a very solid approach.

**Comment by krulunio** [comment]
What if we give fungaloids some fish AI, so they can only move into infected tiles? Another idea, I'm not sure if it isn't already implemented, but some tiles could have chance to resist infection, i.e. dead plants and wood would always become infected, dirt - usually, alive plants - sometimes, concrete and asphalt - rarely. It could reinforce idea of fungus using agents (dead or alive) to spread out, while being rather stationary itself.

**Reply by I-am-Erk** [reply]
Hmm. Having fungaloids only move on infected tiles might work just fine. it would fit with spore clouds, which would then serve the role of extending their reach out.

**Comment by pehamm** [comment]
Could we add mobile agents that spawn at a certain influence and carry some of that influence outwards? Something like slow-moving fungal blossoms that  spawn close to the bloom at a high influence, then start to move out, throwing out spores and becoming a new tower if they reach some destination far enough from the last tower? These could become priority-targets to stop, and ensure that the mycus expands in clear offshoots rather than radially in all directions.

**Reply by I-am-Erk** [reply]
I think that's a reasonable idea, but I would suggest it might be better to save that specific mechanic for triffids - which could use the same mechanic almost exactly but apply it in new ways. Queens are already a triffid thing, the mycus seem more distributed while triffid are more eusocial.

**Reply by pehamm** [reply]
Good points. Triffids do really need some love, currently they do not feel like a faction even remotely comparable to the mycus or the blob. If I had not read the design document I would have thought they were something like a plant version of giant ants, because they function in essentially the same way ingame. But then, this can be tackled once the mycus is less annoying to play with.

**Reply by actual-nh** [reply]
Ping: @Mom-Bun - mentioned on discord meaning to do some work on triffids.

**Comment by Hymore246** [comment]
I've been giving a lot of thought on how to improve the Mycus to make them more fair and engaging for players.  Here are my thoughts:

1. **Mycus attacks the player indirectly with "invisible effects."** Let standing/moving on fungal terrain have a very low chance to cover the player in spores. Add more fungal conditions. They do not appear in the status menu and all have the same "basic" symptoms such as coughing and tasting mushrooms to subvert player knowledge. Spores can cause one of the following conditions: 

    - **Fungal Infection** - Redesigned to only break one limb (arm/leg) at a time but if at least two limbs are broken, it has a chance to target head and torso which instantly kills the player. It can kill but only if ignored and breaking only one limb at a time reduces the chance of the character from becoming "useless." 
    - **Fungal Asthma** - Based on Candidiasis, this condition causes lots of coughing, vomiting, and stamina penalties. It will also cause a small amount of torso damage each time it triggers. It can kill but does so slowly.
    - **Fungal Psychedelic** - Based off Magic Mushrooms, the Mycus got you high! This condition has a few random positive effects such as mood improvement and morale boosts but also has so really bad effects such as hallucinations, body temperature problems, and vomiting. Imagine permanently tripping on shrooms, that is this condition.
    - **Fungal Rash** - Based off Ringworm and Athlete's Foot, this condition is not lethal but it was cause an annoying rash that will cause heavy penalties to your mood and speed for as long as you have it. 

2. **Players need a reason to fight the Mycus.** It is perfectly reasonable to run from the Mycus but that can't be the only thing players ever do. All the work put into Mycus is going to be worthless if players still run the No Fungus mod because Mycus are annoying to deal with and offer nothing to the player. Aside from the Mycus mutation line and Marloss Berries, here are some more ideas to lure players into fighting the Mycus.
    - **New set pieces** - For example, an untouched bunker covered in fungus from a tower that appeared nearby. Another example is a car park with untouched vehicles surrounded by Mycus. If you can get past the Mycus, there are a lot of rewards!
    - **Mycus Penicillin** - Player can use the Mycus to synthesis a powerful antibiotic to heal a number of other illnesses. Alternatively, the Mycus might create this themselves to lure players to them.
    - **Fertilizer** - Fungus in the real life can be used as fertilizer to enhance crops. If harvested and prepared the right way, the Mycus might be able to do the same or work even better than existing fertilizer.
    - **Bioluminescent Fungus** - In real life, sometimes of fungus can glow in the dark. The Mycus might be able to do this too or maybe even do it better and function as a continuous natural light source.

**Reply by Zireael07** [reply]
Antibiotic/fertilizer/lighting sounds like very good reasons (and fairly easy to put in the game unlike coming up with unique "set pieces").

**Reply by actual-nh** [reply]
What is the logic of breaking/disabling a limb?

**Reply by Hymore246** [reply]
Fungus grows inside the player and bursts out. Probably the player version of Fungalization.

**Reply by SariusSkelrets** [reply]
Can we know of what the fertilizer will be made?

Since some are able to be processed into explosives (looking at you, ammonium nitrate), allowing people to make bombs out of something that was process from fungus would make interacting with them a valid survival strategy and make players seek them actively 

I think that the fungal infection should only be inflicted by spore cloud/fungaloid contact. Walking on fungal tiles only to risk a gruesome death as fungus burst out of you doesn’t look fun

What is the amount of coverage/environmental protection needed? It is full body cleansuit and gasmask only to walk on fungal tiles risk-free or some work boots and a sturdy pair of pants would suffice?

**Reply by LyleSY** [reply]
Maybe a fungal war quest chain with some neat rewards?

**Reply by krulunio** [reply]
Seems like good idea for repeatable mission. Could be given by Representative after finishing his main quest line.

**Comment by curstwist** [comment]
The fungal hallucinogenic effect could take the form of mycus propaganda specifically.  A way to sell you on joining them in a full sensory experience

**Reply by LyleSY** [reply]
And give a little joy boost

**Comment by LyleSY** [comment]
I've been thinking about fungal evolutions for a while. I like your points about plague/stealth. What do you think about the darkness of the night? Should that matter? Maybe fungus only grows at night to make it more mysterious and shocking? This would also be a huge performance benefit.

**Reply by I-am-Erk** [reply]
Hmm. it's an intriguing idea but I am not sure about flavour, and I don't think it would help that much with performance. Larger scale changes are needed there.

What it does make me wonder is if we could have it spread primarily underground, forming tunnels a bit like season 2 stranger things, then bursting out with fungus. That would be a huge paradigm shift in how it presents, but make it an extremely different enemy. Or maybe that concept should be used in something else.

**Reply by actual-nh** [reply]
It could also be disrupting anything intact underground - Hub01 mission to fix, perhaps?

**Comment by Hymore246** [comment]
Lets talk set-pieces for Mycus. These are extremely important not only for content but for framing too. After creating my Reddit topic, I chatted with the devs about changing the Mycus to be more reasonable for the player to interact with. Specifically, I wanted to change the Fungaloid spore spamming and Fungal Infection rending characters helpless. I was told they are working as intended and it is because the rest of the Mycus content wasn't in the game yet. So the issue here is that the Fungaloids need to spawn in specific locations to avoid players mindlessly attacking them since they don't know any better. Think Mi-go Slavers belonging to Mi-go Camps and Mi-go Towers.

Here are a few set-pieces I have thought up.

1.  **_____ + Mycus** - Take something already in the game and just add a Mycus to it. Fungal Labs, Fungal Bunkers, etc. This is a simple, but boring way to add Mycus locations to the game.
2. **Hospital Outbreak** - A fungal infection spread through a clueless hospital and most people have died or become fungal zombies. Unlike most Mycus locations, this one doesn't feature a lot of fungal terrain or a fungal tower. If normal zombies are present, the fungal zombies will be hostile to them and vis-versa.
3. **Plants vs Fungus** - The player walks into an all out battle between the Mycus and the Triffid. Both sides are still hostile to the player so they will need to figure out how to get around the battle without everyone attacking them. Depending on how the battle goes, the player might be able to harvest a lot of materials from both sides.
4. **Mycus Greenhouse** - Either by accident or on purpose, Mycus fungus has taken over a greenhouse. Thankfully, the greenhouse is sealed so the Mycus cannot spread BUT the Mycus has converted all the plants inside into some very nice loot (Marloss Berries, Mycus Penicillin, etc) in order to tempt someone into opening the greenhouse and releasing the Mycus. 

These ideas probably need to be fleshed out some more but I think they are a good starting point.

**Reply by krulunio** [reply]
Sounds good, except Mycus Greenhouse, it only takes one broken window to release fungus. 
Plants vs Fungus actually could be ____ vs Fungus.

**Reply by Hymore246** [reply]
Might work better if it was a secure greenhouse in a lab with reinforced glass. Although, Mycus getting into normal greenhouses should be a thing too.

**Comment by I-am-Erk** [comment]
It's mentioned upthread but I should add it to the main post: I think we should strongly consider making mycus spread underground in traversible tunnels that occasionally pop above ground as towers and blooms. This would be (1) awesome, (2) very different from current gameplay giving them a unique edge, (3) less impactful on looting and things, and (4) awesome.

**Reply by actual-nh** [reply]
I'm guessing they'd start spreading through sewers and subways if their tunnels intersected with those?

**Reply by krulunio** [reply]
I think sewers and subways could spawn camps with Marloss Evangelists

**Comment by ActualDio** [comment]
Is there actually a viable way to join fungus with faction control and spread? If a modular system can be made that is robust enough it could even be used to model other biome expansions and effects perhaps? Maybe like wasp nests expanding overtime, forests growing, maybe even modeling decay in citties and other man made structures over time. That could be very useful and powerful if it can be done.

**Comment by Jan-Blasiak** [comment]
Also, to add to a Fungus VS Triffids battle, Triffids could be made to secrete fungicidial gas when passing over Mycus infected tiles, and gas secreted by Triffids could be used to convert fungal trees into normal or **Triffid mutated trees**.
